### PR TITLE
Fixes #2009: Timer Voice minute call not working for persistent timers

### LINKED ISSUE #2009: Timer Voice Miniute Call not working on  V2.0.15
Checked my TX after seeing this reported on RC Groups and the mins are not being called out.

Tom

### PR DESCRIPTION
**Pull request by projectkk2glider**
Timers moved to timer.cpp
Added timer overflow protection
Added timer gtests
THt mode now triggers on THR>10%

Timers code was so complex I decided to write a bunch of gtest for them. All timers related code moved to separate file. Unfortunately this move hides real fix of issue #2009, but according to tests it should be fixed now. Please review and test before the merge. We must be careful on `master` branch. If this pull is accepted, I will port it to the `next`.

### CODE REVIEW

File: radio/src/timers.cpp [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+
+void evalTimers(int16_t throttle, uint8_t tick10ms)
+{
+  for (uint8_t i=0; i<MAX_TIMERS; i++) {
+    int8_t tm = g_model.timers[i].mode;
+    uint16_t tv = g_model.timers[i].start;
+    TimerState * timerState = &timersStates[i];
+
+    if (tm) {
+      if ((timerState->state == TMR_OFF) && (tm != TMRMODE_THR_TRG)) {
```
Comment (projectkk2glider) [inline radio/src/timers.cpp]: First change, leave state TMR_OFF if timer mode == TMRMODE_THR_TRG. Needed to handle THt persistent timer start.

File: radio/src/timers.cpp [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+      }
+
+      if (tm == TMRMODE_THR_REL) {
+        timerState->cnt++;
+        timerState->sum += throttle;
+      }
+
+      if ((timerState->val_10ms += tick10ms) >= 100) {
+        if (timerState->val == TIMER_MAX) break;
+        if (timerState->val == TIMER_MIN) break;
```
Comment (projectkk2glider) [inline radio/src/timers.cpp]: Added timer wrap-arround stopper. This was another (unrelated) issue, but I decided to fix it. It increases stock FLASH size.

File: radio/src/timers.cpp [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+          // we can't rely on (throttle || newTimerVal > 0) as a detection if timer should be running
+          // because having persistent timer brakes this rule
+          if ((throttle > THR_TRG_TRESHOLD) && timerState->state == TMR_OFF) {
+            timerState->state = TMR_RUNNING;  // start timer running
+            timerState->cnt = 0;
+            timerState->sum = 0;
+            TRACE("Timer[%d] THr triggered", i);
+          }
+          if (timerState->state != TMR_OFF) newTimerVal++;
+        }
```
Comment (projectkk2glider) [inline radio/src/timers.cpp]: New code for THt handling

**Comment by bsongis** [comment]
Thanks for this one!

**Pull request merged**